marimbabot_planning/src/planning.cpp

```diff
@@ -263,35 +266,36 @@ moveit::planning_interface::MoveGroupInterface::Plan Planning::hit_note(
  * @param start_state
  * @param points
  * @return moveit::planning_interface::MoveGroupInterface::Plan
+ * @return ros::Duration
 **/
-moveit::planning_interface::MoveGroupInterface::Plan Planning::hit_notes(
+std::tuple<moveit::planning_interface::MoveGroupInterface::Plan, ros::Duration> Planning::hit_notes(
     const moveit_msgs::RobotState& start_state,
     std::vector<CartesianHitSequenceElement> points)
-{
+{  
     // Assert that there is at least one hit_point with an nice error message
     assert(points.size() > 0 && "There must be at least one hit_point");
 
     std::vector<ros::Duration> note_hit_times;
+
     // Calculate hit trajectory
-    auto hit_plan = hit_note(
-        start_state,
-        points.front()
-        );
+    std::tuple<moveit::planning_interface::MoveGroupInterface::Plan, ros::Duration> result = hit_note(start_state, points.front());
```

Comment:
Modern C++ supports unpacking. Sadly we don't use C++17, which supports python style unpacking, but we can still simplify the code to:

```c++
moveit::planning_interface::MoveGroupInterface::Plan hit_plan;
ros::Duration approach_time_to_first_note_hit;
std::tie(hit_plan, approach_time_to_first_note_hit) = hit_note(start_state, points.front());
```

in C++17 this would be possible:

```C++
auto [hit_plan, approach_time_to_first_note_hit] = hit_note(start_state, points.front());
```

Reply:
yes I stumpled accross that as well